base/src/main/java/io/spine/validate/constraint/ExternalConstraints.java

```diff
@@ -0,0 +1,187 @@
+/*
+ * Copyright 2019, TeamDev. All rights reserved.
+ *
+ * Redistribution and use in source and/or binary forms, with or without
+ * modification, must retain the above copyright notice and the following
+ * disclaimer.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package io.spine.validate.constraint;
+
+import com.google.common.base.Splitter;
+import com.google.common.collect.ImmutableSet;
+import com.google.protobuf.DescriptorProtos;
+import io.spine.annotation.Internal;
+import io.spine.logging.Logging;
+import io.spine.type.KnownTypes;
+import io.spine.type.MessageType;
+import org.slf4j.Logger;
+
+import java.io.Serializable;
+import java.util.Collection;
+import java.util.Set;
+import java.util.function.Predicate;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+import static com.google.common.collect.ImmutableSet.toImmutableSet;
+import static com.google.common.collect.Sets.newHashSetWithExpectedSize;
+import static io.spine.util.Exceptions.newIllegalArgumentException;
+
+/**
+ * A collection of {@linkplain ExternalMessageConstraint external constrains} known to
+ * the application.
+ *
+ * <p>During initialization of this class, definitions of external constraints are verified.
+ * If an invalid constraint was found, a runtime exception will be thrown.
```

Comment:
`If an invalid constraint is found, a runtime exception is thrown.`

Also, I would explain what you mean by "invalid" and "verified".